sdk/dotnet/Pulumi/Config.cs

```diff
@@ -0,0 +1,361 @@
+         *
+         * @param key The key to lookup.
+         * @param opts An options bag to constrain legal values.
+         */
+        /// <summary>
+        /// Loads an optional configuration value, as a number, by its key, or null if it doesn't exist.
+        /// If the configuration value isn't a legal number, this function will throw an error.
+        /// </summary>
+        public int? GetInt32(string key, Int32Options? opts = null)
+        {
+            var v = this.Get(key);
+            if (v == null)
+            {
+                return null;
+            }
+
+            if (!int.TryParse(v, out var result))
+            {
+                throw new ConfigTypeException(this.FullKey(key), v, "Int32");
+            }
+
+            if (opts != null)
+            {
+                if (opts.Min != null && result < opts.Min)
+                {
+                    throw new ConfigRangeException(this.FullKey(key), result, opts.Min, null);
+                }
+                else if (opts.Max != null && result > opts.Max)
+                {
+                    throw new ConfigRangeException(this.FullKey(key), result, null, opts.Max);
+                }
+            }
+            return result;
+        }
+
+        /*
+         * getSecretInt32 loads an optional configuration value, as a number, by its key, marking it as a secret
+         * or null if it doesn't exist.
+         * If the configuration value isn't a legal number, this function will throw an error.
+         *
+         * @param key The key to lookup.
+         * @param opts An options bag to constrain legal values.
+         */
+        /// <summary>
+        /// loads an optional configuration value, as a number, by its key, marking it as a secret
+        /// or null if it doesn't exist.
+        /// If the configuration value isn't a legal number, this function will throw an error.
+        /// </summary>
+        public Output<int>? GetSecretInt32(string key, Int32Options? opts = null)
+        {
+            var v = this.GetInt32(key, opts);
+            if (v == null)
+            {
+                return null;
+            }
+
+            return MakeSecret(v.Value);
+        }
+
+        //    /**
+        //     * getObject loads an optional configuration value, as an object, by its key, or null if it doesn't exist.
+        //     * This routine simply JSON parses and doesn't validate the shape of the contents.
+        //     *
+        //     * @param key The key to lookup.
+        //     */
+        //    public getObject<T>(string key): T | null {
+        //        var v: string | null = this.Get(key);
+        //        if (v == null) {
+        //    return null;
+        //}
+        //        try {
+        //    return < T > JSON.parse(v);
+        //}
+        //        catch (err) {
+        //    throw new ConfigTypeException(this.FullKey(key), v, "JSON object");
+        //}
+        //}

```

Comment:
didn't implement this bit yet.  not sure the main use cases for this in .net.

Reply:
FYI, these are needed for https://github.com/pulumi/pulumi/pull/3342

Reply:
good to know.  i'll make sure we support these.